### PR TITLE
Add additional classification (F1, Hamming loss) and regression (MSE) metrics

### PR DESCRIPTION
**Pull request by matejklemen**
##### Description of Changes

Adds a few new classification (F1 score, Hamming loss) and regression (MSE) metrics + tests. I've also changed RMSE so that it reuses the MSE functionality.     
Partially implements #12.

##### Includes

- [X] Code changes
- [X] Tests
- [ ] Documentation

##### Note
I'm not sure if the checks for inputs (`y` and `yPred`) being of equal length are necessary (not sure how rigorously input should be checked). If they stay in, they should probably also be added to other metrics.

### CODE REVIEW

**Comment by inejc** [comment]
Thanks for the PR @matejklemen. I'll check it out when back from vacation.

File: src/main/scala/io/picnicml/doddlemodel/metrics/ClassificationMetrics.scala [hunk 1]
```diff
@@ -42,4 +42,30 @@ object ClassificationMetrics {
...
+
+    override lazy val higherValueIsBetter: Boolean = true
+
+    override def apply(y: Target, yPred: Target): Double = {
+      require(y.length == yPred.length, "Target vectors need to be of equal length")
+      require(numberOfTargetClasses(y) == 2, "F1 score is defined for a binary classification task")
+      val prec = precision(y, yPred)
+      val rec = recall(y, yPred)
+
+      2 * (prec * rec) / (prec + rec)
```
Comment (inejc) [inline src/main/scala/io/picnicml/doddlemodel/metrics/ClassificationMetrics.scala]: Really nitpicking here but I usually like to write constants that are converted to doubles as `2.0` for example, at least my mental model is that as soon as I see the double format I know that the evaluated expression will be of the type `Double`. The same as you did for the Hamming loss.

File: src/main/scala/io/picnicml/doddlemodel/metrics/package.scala [hunk 2]
```diff
@@ -15,6 +16,8 @@ package object metrics {
   lazy val accuracy: Metric = Accuracy
   lazy val precision: Metric = Precision
   lazy val recall: Metric = Recall
+  lazy val f1score: Metric = F1Score
```
Comment (inejc) [inline src/main/scala/io/picnicml/doddlemodel/metrics/package.scala]: `f1score` should probably be called `f1Score` 🤔 ?

**Comment by matejklemen** [comment]
Sure, I can fix all of these things - will do it once I find some time :slightly_smiling_face:

**Comment by inejc** [comment]
LGTM, thanks for your contribution @matejklemen.

**Pull request merged**